src/main/topLevelPages/myDashboard/myMemberships/MyMembershipsSpaceCard.tsx

```diff
@@ -0,0 +1,103 @@
+import RouterLink from '../../../../core/ui/link/RouterLink';
+import { BlockSectionTitle, BlockTitle, Caption } from '../../../../core/ui/typography';
+import { MembershipProps } from './MyMembershipsDialog';
+import { CommunityRole } from '../../../../core/apollo/generated/graphql-schema';
+import webkitLineClamp from '../../../../core/ui/utils/webkitLineClamp';
+
+interface MyMembershipsSpaceCardProps {
+  displayName: string;
+  tagline?: string;
+  avatar?: string;
+  url: string;
+  roles?: CommunityRole[];
+  level: number;
+  subspaces: MembershipProps[] | undefined;
+}
+
+const VISIBLE_COMMUNITY_ROLES = [CommunityRole.Admin, CommunityRole.Lead];
+
+const MyMembershipsSpaceCard = ({
+  displayName,
+  tagline,
+  avatar,
+  url,
+  roles,
+  level,
+  subspaces,
+}: MyMembershipsSpaceCardProps) => {
+  const { t } = useTranslation();
+  const isMobile = useMediaQuery<Theme>(theme => theme.breakpoints.down('sm'));
+  const [isExpanded, setIsExpanded] = useState(false);
+  const toggleExpanded = () => setIsExpanded(wasExpanded => !wasExpanded);
+
+  const communityRoles = roles?.filter(role => VISIBLE_COMMUNITY_ROLES.includes(role)).sort();
+
+  return (
+    <>
+      <GridItem columns={12}>
+        <Gutters flexDirection="row" paddingY={isMobile ? 0.5 : 0} paddingLeft={level * 5} paddingRight={0} marginY={0}>
+          <BadgeCardView
+            visual={
+              <Avatar
+                src={avatar || defaultCardBanner}
+                aria-label="Space avatar"
+                alt={t('common.avatar-of', { space: displayName })}
+              >
+                {displayName[0]}
+              </Avatar>
+            }
+            component={RouterLink}
+            to={url}
+            sx={{ flexGrow: 1 }}
+          >
+            <BlockTitle sx={isMobile ? webkitLineClamp(2) : undefined}>{displayName}</BlockTitle>
+            <BlockSectionTitle sx={isMobile ? webkitLineClamp(2) : undefined}>{tagline}</BlockSectionTitle>
+            {isMobile && (
+              <Caption color="primary">
+                {communityRoles?.map(role => t(`common.enums.communityRole.${role}` as const)).join(', ')}
+              </Caption>
+            )}
+          </BadgeCardView>
+          <Gutters flexDirection="row">
+            {!isMobile && (
+              <Caption color="primary">
+                {communityRoles?.map(role => t(`common.enums.communityRole.${role}` as const)).join(', ')}
+              </Caption>
+            )}
+            <Button
+              onClick={toggleExpanded}
+              endIcon={isExpanded ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+              sx={{ visibility: subspaces?.length ? 'visible' : 'hidden' }}
+              area-label={isExpanded ? t('buttons.collapse') : t('buttons.expand')}
+            />
+          </Gutters>
+        </Gutters>
+      </GridItem>
+      {isExpanded &&
+        subspaces?.map(subspace => (
+          <MyMembershipsSpaceCard
```

Comment:
This nesting looks dangerous, but it works well for our structure. 

I suggest rethinking the name of the component as it's not just a Space card but as it renders its children subspaces, it could be something like MyMembershipsSpaceTree.